.golangci.yml

```diff
@@ -1,39 +1,41 @@
 linters-settings:
   gofumpt:
-    lang-version: "1.18"
+    lang-version: '1.19'
 
   misspell:
     locale: US
 
+  staticcheck:
+    go: '1.19'
+    checks: ['all', '-ST1005', '-ST1000', '-SA4000', '-SA9004', '-SA1019', '-SA1008', '-U1000', '-ST1016']
+
 linters:
   disable-all: true
   enable:
-    - typecheck
+    - durationcheck
+    - gocritic
+    - gofmt
+    - gofumpt
     - goimports
-    - misspell
+    - gomodguard
     - govet
-    - revive
     - ineffassign
-    - gomodguard
-    - gofmt
+    - misspell
+    - revive
+    - staticcheck
+    - tenv
+    - typecheck
     - unconvert
     - unused
-    - gocritic
-    - gofumpt
-    - tenv
-    - durationcheck
 
 issues:
   exclude-use-default: false
   exclude:
-      - should have a package comment
-      - error strings should not be capitalized or end with punctuation or a newline
-      # todo fix these when we get enough time.
-      - "singleCaseSwitch: should rewrite switch statement to if statement"
-      - "unlambda: replace"
-      - "captLocal:"
-      - "ifElseChain:"
-      - "elseif:"
-
-service:
-  golangci-lint-version: 1.43.0 # use the fixed version to not introduce new linters unexpectedly
+    - should have a package comment
+    - error strings should not be capitalized or end with punctuation or a newline
+    # todo fix these when we get enough time.
+    - 'singleCaseSwitch: should rewrite switch statement to if statement'
+    - 'unlambda: replace'
+    - 'captLocal:'
+    - 'ifElseChain:'
+    - 'elseif:'
```

Comment:
Not really sure why these are excluded. Oh well, we don't have to fix that here, tough.

Reply:
It seems they will cause a bigger diff. Let me get rid of them separately and refactor relevant places.